cassdegrees/ui/views.py

```diff
@@ -1,8 +1,66 @@
+from django.http import HttpResponse
 from django.shortcuts import render
-import os
+import requests
 
 
 # Create your views here.
 # I added a very simple sample request handler, this is very simple and all it does is load index.html from templates.
 def index(request):
     return render(request, 'index.html', context={})
+
+
+# I went through this tutorial to create the form html file and this view:
+# https://docs.djangoproject.com/en/2.2/topics/forms/
+# Hope this serves as an inspiration for when we make proper views and functions to submit course information
+def sampleform(request):
+    # If POST request, redirect the received information to the backend:
+    if request.method == 'POST':
+        model_api_url = 'http://127.0.0.1:8000/api/sample/'
```

Comment:
Hmmm.... Not sure I like the idea of bouncing stuff around via HTTP internally - no way to directly talk to the API?

Reply:
Not that I know of, the API from the Django REST Framework is built to accept incoming POST, PUT, PATCH and DELETE requests so I used them.

Reply:
The problem is overhead + port changes + etc here. We are effectively handling double HTTP requests here now.

Reply:
Keep in mind that our UI and API are considered to be separate apps, so importing the API related views to the UI application doesn't make much sense. Since they are separate apps, it makes sense to submit HTTP requests to the API endpoint.

My idea here was to receive the data from the user input from the form, package it nicely in the UI app to a format the API (api/sample/) will accept and then submit a HTTP request to the API endpoint.

Have a look at the link below:
https://stackoverflow.com/questions/11663945/calling-a-rest-api-from-django-view

Regarding your point on overhead, could you please clarify what kind of overhead (and where) you are talking about?

Hard coding URL is a bad idea that I do admit, but we have no better method of dealing with that as of this moment and is not relevant to this pull request.

Double HTTP requests are of little concern (minimal performance impact), the data must be transferred to the API regardless of method (whether function calls or HTTP request).

However, I am open to suggestions, please feel free to let me know if you find a better method of going about this.

Reply:
In terms of overhead, for each of these API requests:
- A new TCP connection has to be created, including operating system overhead (lookup, routing, etc), which is a non-zero cost for network connections.
- HTTP request + response parsing
- Domain filtering (i.e ALLOWED_HOSTS)

If we cannot talk between the API and UI directly, would unix sockets also be a solution?

Reply:
I am not familiar at all with Unix sockets, but I'm not sure if it would be a solution here as "A UNIX socket is an inter-process communication mechanism that allows bidirectional data exchange between processes running on the same machine." - https://serverfault.com/questions/124517/whats-the-difference-between-unix-socket-and-tcp-ip-socket.

UI and API are not actual standalone processes but are simply separate web-apps. While I said you can't import things from API to UI, you can but it is not good practice to be doing that and I am not aware of any Python functions of ListAPIView (api/sample/) that you can call to insert data.

But if we choose to go down this route, this means that we will have to create listeners, learn to understand Unix socket data transmissions etc for some gain that in reality, no one will notice. In my opinion, any benefit gained by going this route will be far outweighed by the things I have said in the previous sentence.

Reply:
That is fair then. Fairly disappointing, but hopefully this doesn't become an issue.